pipelines/pom.xml

```diff
@@ -108,6 +108,12 @@
       <version>${logback.version}</version>
       <scope>runtime</scope>
     </dependency>
+    <dependency>
```

Comment:
Please add an explanation why this is needed; I don't fully figure why Maven is picking a different version while `logback-core` is a direct dependency of `logback-classic`.

Reply:
I had a deeper look at this, here is the explanation why the maven was picking a different version.

In the `pipelines/controller/pom.xml` file, we have a declaration for inheriting the `dependencyManagement` section of the artifact `org.springframework.boot:spring-boot-dependencies` over [here](https://github.com/google/fhir-data-pipes/blob/2fe486e36aecf77194b5f7df9d7dfdde48cf487d/pipelines/controller/pom.xml#L37). And in this artifact it is defined to use the version `1.4.11` for `logback-classic`, `logback-core` and `logback-access`, check [here](https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/3.1.5/spring-boot-dependencies-3.1.5.pom) for details.

Whereas, in our main `pipelines/pom.xml` file we have added the dependency for `logback-classic` and specified version for it. However, since no version is specified for `logback-core` the version that is specified in the `dependencyManagement` section via the `org.springframework.boot:spring-boot-dependencies` artifact gets applied. If there was no mention of version in the `dependencyManagement` section then the version would have been derived from the parent dependency `logback-core`.

Reply:
Somehow the `mvn dependency:tree` was not very clear on how the versions were getting resolved. The `mvn help:effective-pom` command helped me to analyze a bit more.

Reply:
I see, thanks for getting to the bottom of this and the explanation.